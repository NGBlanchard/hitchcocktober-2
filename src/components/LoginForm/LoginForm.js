import React, { Component } from "react";
import ApiService from "../../services/api-service";
import TokenService from "../../services/token-service";
import Context from "../../Context";

import './LoginForm.css'

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: (id) => {}
  };

  state = { error: null };
  static contextType = Context;

  handleAuth = e => {
    e.preventDefault();
    this.setState({ error: null });
    const { user_name, password } = e.target;
    ApiService.postLogin({
      user_name: user_name.value,
      password: password.value
    })
      .then(res => {
        user_name.value = "";
        password.value = "";
        TokenService.saveAuthToken(res.authToken);
        TokenService.setUser(res.user.user_name);
        TokenService.setDate(res.user.date_created);
        TokenService.setUserId(res.user.id);
        this.props.onLoginSuccess(res.user.id);
      })
      .catch(res => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;
    return (
      <React.Fragment>
        <form className="LoginForm" onSubmit={this.handleAuth}>
          <span role="alert">{error && <p className="red">{error}</p>}</span>
          <section className="user_name">
            <label htmlFor="LoginForm__user_name">Username</label>
            <input name="user_name" id="LoginForm__user_name"/>
          </section>
          <section className="password">
            <label htmlFor="LoginForm__password">Password</label>
            <input
              name="password"
              type="password"
              id="LoginForm__password"
            />
          </section>
          <button type="submit">Login</button>
        </form>
      </React.Fragment>
    );
  }
}
