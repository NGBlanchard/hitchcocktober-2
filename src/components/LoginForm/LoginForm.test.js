import LoginForm from "./LoginForm";
import React from "react";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<LoginForm />", () => {
  it("Renders without crashing", () => {
    mount(<LoginForm />);
  });
  it("renders Login Form by default", () => {
    const wrapper = mount(<LoginForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
