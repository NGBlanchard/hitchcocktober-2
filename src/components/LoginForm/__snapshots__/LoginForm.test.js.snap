// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders Login Form by default 1`] = `
<LoginForm
  onLoginSuccess={[Function]}
>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <span
      role="alert"
    />
    <section
      className="user_name"
    >
      <label
        htmlFor="LoginForm__user_name"
      >
        Username
      </label>
      <input
        id="LoginForm__user_name"
        name="user_name"
      />
    </section>
    <section
      className="password"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <input
        id="LoginForm__password"
        name="password"
        type="password"
      />
    </section>
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</LoginForm>
`;
